### PR TITLE
[Modules] Update Network FrontDoors to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Update Network FrontDoors to new dependency approach

| Pipeline |
| - |
| [Network: Frontdoors](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.frontdoors.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Network/frontDoors/.test/common/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,135 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes')
```
Comment (MrMCake) [inline modules/Microsoft.Network/frontDoors/.test/common/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes')
```

File: modules/Microsoft.Network/frontDoors/.test/min/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,116 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes')
```
Comment (MrMCake) [inline modules/Microsoft.Network/frontDoors/.test/min/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes')
```

File: modules/Microsoft.Network/frontDoors/.test/common/deploy.test.bicep [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+// =========== //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+module resourceGroupResources 'dependencies.bicep' = {
```
Comment (eriqua) [inline modules/Microsoft.Network/frontDoors/.test/common/deploy.test.bicep:27]: The dependency is not used. Should it be added?

Comment (MrMCake) [reply modules/Microsoft.Network/frontDoors/.test/common/deploy.test.bicep:27]: oha. Added

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**